### ISSUE TITLE
MSH leaving zombie Java process in docker container

### ISSUE THREAD

**Issue by j4c3** [gekware/minecraft-server-hibernation#146]
### Current Behavior (🐛 Bug)

Msh starts the server fine.  Players leave, and it shuts down, as expected:

```
2022/04/13 01:31:11 [info  *** ] 0 online players - method for player count: server info
2022/04/13 01:31:11 [serv  **  ] ms command: stop       (origin: StopMS)
2022/04/13 01:31:11 [serv  **  ] [18:31:11 INFO]: Stopping the server
2022/04/13 01:31:11 [info  *** ] MINECRAFT SERVER IS STOPPING!
...
2022/04/13 01:31:15 [serv  **  ] [18:31:15 INFO]: ThreadedAnvilChunkStorage: All dimensions are saved
2022/04/13 01:31:15 [serv  **  ] [18:31:15 INFO]: Flushing Chunk IO
2022/04/13 01:31:15 [serv  **  ] [18:31:15 INFO]: Closing Thread Pool
2022/04/13 01:31:15 [serv  **  ] 2022-04-09 18:31:15,353 Log4j2-AsyncAppenderEventDispatcher-1-Async WARN Advanced terminal features are not available in this environment
2022/04/13 01:31:15 [serv  **  ] [18:31:15 INFO]: Closing Server
2022/04/13 01:31:15 [info  *** ] MINECRAFT SERVER IS OFFLINE!
```

But, running a PS -aux, or PS within the container I see:

`mc        330547 72.1  0.0      0     0 ?        Z    16:32  32:48 [java] <defunct>`

I am running images based off openJDK17-slim, and openJDK19-slim.  Both leave the Java zombies. 

### Expected Behavior

Stopping the MC server, then releasing the handle that is monitoring or connected to the java process?

### Steps to Reproduce

Starting it up, then letting MSH shutdown the Minecraft server. 

### MSH-Version

2.4.6

### Operating system

Ubuntu 20.04.4

### Minecraft Version

1.18.2

### Minecraft Server Software

Paper

### Is there an existing issue for this?

- [X] I have searched the existing issues before opening this issue.
- [X] I have provided all relevant details.

**Comment by gekigek99** [comment]
thanks for reporting!

1) can you test starting and stopping a normal minecraft server? (in docker container but without msh, just good old `java -Xmx1024M -Xms1024M -jar minecraft_server.1.18.2.jar nogui` ?)
2) have you tryed the naked msh executable? without ptero egg/docker container?

**Comment by j4c3** [comment]
Just tested:
1.  Starting and stopping the jar in docker directly works as expected, no zombies. 
2.  Running msh outside of docker produces the same result for me.  A zombie is left when msh shuts down the server. 

I don't know Go, but this appears to be a normal issue in Unix land, where a terminated child process is always a zombie, until it is processed by its parent. 

There is a Go function in the syscall package that seems to relate to this, wait4 (which accepts a PID) and I think that would handle it.  

I'm not sure where would be best, though.  In waitForExit() in servctrl-cmd.go, you have the terminal, and could probably get the PID, but syscall isn't there.  In progmgr.go, you're already listening for interrupts, and I think this situation raises a SIGCHLD, so that would make sense, but it may be harder to get the PID there.

**Comment by gekigek99** [comment]
>There is a Go function in the syscall package that seems to relate to this, wait4 (which accepts a PID) and I think that would handle it.
>
>I'm not sure where would be best, though. In waitForExit() in servctrl-cmd.go, you have the terminal, and could probably get the PID, but syscall isn't there. In progmgr.go, you're already listening for interrupts, and I think this situation raises a SIGCHLD, so that would make sense, but it may be harder to get the PID there.

thanks for researching into that, I also found the stuff you described but they quite low level...
fortunately they should be already implementd in this function: try adding `ServTerm.cmd.Wait()` at line 296 here:

https://github.com/gekware/minecraft-server-hibernation/blob/dev/lib/servctrl/servctrl-cmd.go#L294-L296

can you check if that is the case?

p.s. if you dig down in the source code of `(*exec.Cmd).Wait()` you will find the implementation of `wait4` and `SIGCHLD`

**Comment by j4c3** [comment]
Just tested, and adding that line at 296 does seem to resolve the issue!

**Comment by gekigek99** [comment]
Thank you for reporting and researching into the issue, good job!

fixed in e2db861a1d51b80a3a615097d3c141512d565ada
